ic-utils/src/interfaces/wallet.rs

```diff
@@ -88,9 +98,55 @@ where
 }
 
 /// A wallet canister interface, for the standard wallet provided by DFINITY.
-/// This interface implement most methods conveniently for the user.
+/// This interface implements most methods conveniently for the user.
 pub struct Wallet;
 
+#[derive(CandidType, Deserialize)]
+pub struct BalanceResult {
+    pub amount: u64,
+}
+
+#[derive(CandidType, Deserialize)]
+pub struct ReceiveResult {
+    pub accepted: u64,
+}
+
+#[derive(CandidType, Deserialize)]
+pub struct CreateResult {
+    pub canister_id: Principal,
+}
+
+#[derive(Deserialize)]
+pub struct CallResult {
+    pub r#return: Vec<u8>,
+}
+
+/// Due to https://github.com/dfinity/candid/issues/148 we need to manually
+/// implement CandidType trait (for now).
+/// TODO: reuse derive(CandidType) once the issue above is fixed.
```

Comment:
going to remove this once https://github.com/dfinity/agent-rs/pull/100 is merged into `next`